external-crates/move/crates/move-package/src/resolution/dependency_graph.rs

```diff
@@ -1235,15 +1231,43 @@ impl DependencyGraph {
             self.write_dependencies_to_lock(*id, &mut writer)?;
         }
 
-        writer.flush()?;
-        std::mem::drop(writer);
+        let mut dependencies = None;
+        let mut dev_dependencies = None;
+        let mut packages = None;
+        if !writer.is_empty() {
+            let toml = writer.parse::<toml_edit::Document>()?;
+            if let Some(value) = toml.get("dependencies").and_then(|v| v.as_value()) {
+                dependencies = Some(value.clone());
+            }
+            if let Some(value) = toml.get("dev-dependencies").and_then(|v| v.as_value()) {
+                dev_dependencies = Some(value.clone());
+            }
+            packages = toml["move"]["package"].as_array_of_tables().cloned();
+        }
```

Comment:
This takes the dependency graph info as a string (in `writer`) and parses it with `toml_edit` to obtain TOML values. Those TOML values preserve the hand-rolled formatting for the most part, which is the reason for this dance.

Reply:
Can these indexing operations for `["move"]` and `["package"]` fail? (Do we have a test for that?)

Reply:
It's a good call out. There's an invariant here in the string construction before this code: if `writer` is a non-empty string, it means `dependencies` or `dev-dependencies` exist. If these exist, `[[move.package]]` exists.

In `get_graph`, parsing `dependencies` or `dev-dependencies` without `[[move.package]]` fails, so there's no way to trigger this with an incomplete `Move.lock`.

I have converted it to a safe access anyway. So that, if the invariant is broken we'll see snapshot tests reflect the issue, rather than a index panic.